pkg/meta/redis.go

```diff
@@ -254,20 +278,104 @@ func newRedisMeta(driver, addr string, conf *Config) (Meta, error) {
 		}
 	}
 
+	// Setup Redis meta
 	m := &redisMeta{
-		baseMeta: newBaseMeta(addr, conf),
-		rdb:      rdb,
-		prefix:   prefix,
+		baseMeta:          newBaseMeta(addr, conf),
+		rdb:               rdb,
+		prefix:            prefix,
+		clientCache:       clientCache,
+		clientCacheBcast:  true, // Always use BCAST mode for simplicity
+		clientCacheSize:   clientCacheSize,
+		clientCacheExpiry: clientCacheExpiry,
+	}
+
+	// Initialize LRU caches if client-side caching is enabled
+	if clientCache {
+		var err error
+		m.inodeCache, err = lru.New[Ino, *Attr](clientCacheSize)
+		if err != nil {
+			return nil, fmt.Errorf("create inode cache: %s", err)
+		}
+
+		entryCache, err := lru.New[string, struct {
+			ino  Ino
+			attr Attr
+		}](clientCacheSize)
+		if err != nil {
+			return nil, fmt.Errorf("create entry cache: %s", err)
+		}
+		m.entryCache = entryCache
+
+		readCache, err := lru.New[readCacheKey, []*slice](clientCacheSize)
+		if err != nil {
+			return nil, fmt.Errorf("create read cache: %s", err)
+		}
+		m.readCache = readCache
 	}
+
 	m.en = m
 	m.checkServerConfig()
+
+	// We'll initialize the client-side caching after session creation
+	if clientCache {
+		// Only prepare the caches, actual setup will be done after session creation
+		m.setupCachedMethods()
+		if clientCacheExpiry > 0 {
+			logger.Infof("Redis client-side caching prepared with size %d MB (%d entries) and expiry %s",
+				clientCacheSizeMB, clientCacheSize, clientCacheExpiry)
+		} else {
+			logger.Infof("Redis client-side caching prepared with size %d MB (%d entries) and infinite expiry",
+				clientCacheSizeMB, clientCacheSize)
+		}
+	}
+
 	return m, nil
 }
 
 func (m *redisMeta) Shutdown() error {
+	if m.clientCache {
+		m.shutdownClientSideCaching()
+	}
 	return m.rdb.Close()
 }
 
+// Override NewSession to initialize client-side cache after session is created
+func (m *redisMeta) NewSession(record bool) error {
+	// First, create the session normally
+	err := m.baseMeta.NewSession(record)
+	if err != nil {
+		return err
+	}
+
+	// Now that we have a valid session, setup client-side caching if enabled
+	if m.clientCache {
+		err = m.setupClientSideCaching(m.clientCacheExpiry)
+		if err != nil {
+			logger.Warnf("Failed to setup client-side caching: %v", err)
+			m.clientCache = false
+		} else {
+			if m.clientCacheExpiry > 0 {
+				logger.Infof("Redis client-side caching enabled with expiry %s", m.clientCacheExpiry)
+			} else {
+				logger.Infof("Redis client-side caching enabled with infinite expiry")
+			}
+
+			// Preload the first 10000 inodes into the cache in a safe background goroutine
+			// Wrap with a recovery function to prevent any panics from affecting the mount process
+			go func() {
+				defer func() {
+					if r := recover(); r != nil {
+						logger.Errorf("Recovered from panic in cache preloading goroutine: %v", r)
+					}
+				}()
+				m.preloadInodeCache(10000)
```

Comment:
there should be a option for this, otherwise it could overload the Redis when there are many clients.